includes/admin/MessagesRepository.php

```diff
@@ -21,30 +25,68 @@ class MessagesRepository
     public function __construct()
     {
         add_action('admin_notices', [$this, 'showMessages']);
+        add_action('wp_ajax_dismissNotice', [$this, 'dismissNotice']);
     }
 
     /**
-     * @param  string $message
-     * @param  string $level
-     * @param  bool   $onAllPages
+     * @param  string      $message
+     * @param  string      $level
+     * @param  string|null $messageId
+     * @param  array       $onPages
      */
-    public function addMessage(string $message, string $level, bool $onAllPages = false): void
+    public function addMessage(string $message, string $level, ?string $messageId, array $onPages = []): void
     {
         $this->messages[] = [
-            'message'    => $message,
-            'level'      => $level,
-            'onAllPages' => $onAllPages,
+            'message'   => $message,
+            'messageId' => $messageId ?? null,
+            'level'     => $level,
+            'onPages'   => $onPages,
         ];
     }
 
     public function showMessages(): void
     {
         foreach ($this->messages as $message) {
-            if (! $message['onAllPages'] && ! WCMYPA_Settings::isViewingOwnSettingsPage()) {
-                continue;
+            if ($this->shouldMessageBeShown($message)) {
+                $isDismissible = $message['messageId'] ? 'is-dismissible' : '';
+                echo sprintf(
+                    '<div class="notice myparcel-dismiss-notice notice-%s %s"><p>%s</p></div>',
+                    $message['level'],
+                    $isDismissible,
+                    $message['message']
+                );
             }
+        }
+    }
+
+    private function shouldMessageBeShown(array $message): bool
+    {
+        $messageAlreadyShown = in_array($message['messageId'], get_option('myparcel_notice_dismissed'), true);
+        $currentPage         = get_current_screen();
+
+        if ($messageAlreadyShown) {
+            return false;
+        }
+        if (in_array($currentPage->id, $message['onPages'], true)) {
+            return true;
+        }
 
-            echo sprintf('<div class="notice notice-%s"><p>%s</p></div>', $message['level'], $message['message']);
+        if (empty($message['onPages'])) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function dismissNotice(): void
+    {
+        $messageArray = get_option('myparcel_notice_dismissed', []);
+
+        foreach ($this->messages as $message) {
+            if (! in_array($message['messageId'], $messageArray, true)) {
```

Comment:
Graag de happy path op het eind, zodat hij niet indented is en je (hier toevallig) geen negated condition gebruikt, dus zoiets:

```js
if (true) {
  continue;
}

codes...
```

ipv 
```js
if (false) {
  codes...
}
```

_ProTip: Ga in phpstorm op `if (` staan, druk op option+enter en kies "Invert 'if' statement"_